### PR TITLE
chore: 🤖 enable more package_Json side effects tests

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_implicit_main/artifacts.snap [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	console.log('TEST FAILED');
+});
+
+// src/require-demo-pkg.js
+init_index_module(),__toCommonJS(index_module_ns);
+
+// src/entry.js
+init_index_module();
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_implicit_main/artifacts.snap:30]:
```js
================================================================================
TestPackageJsonSideEffectsArrayKeepMainImplicitMain
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index-main.js
var index_main_exports = {};
__export(index_main_exports, {
  foo: () => foo
});
var foo;
var init_index_main = __esm({
  "Users/user/project/node_modules/demo-pkg/index-main.js"() {
    foo = 123;
    console.log("this should be kept");
  }
});

// Users/user/project/src/entry.js
init_index_main();

// Users/user/project/src/require-demo-pkg.js
init_index_main();

// Users/user/project/src/entry.js
console.log("unused import");

```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_implicit_module/artifacts.snap [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_implicit_module
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_implicit_module/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsArrayKeepMainImplicitModule
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

**Review by hyf0 — DISMISSED** [review]

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_use_module/artifacts.snap [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_use_module
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_main_use_module/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsArrayKeepMainUseModule
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_module_implicit_module/artifacts.snap [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## src_entry_js.mjs
+
+```js
+// node_modules/demo-pkg/index-module.js
+console.log('this should be kept');
+
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_module_implicit_module/artifacts.snap:16]:
```js
================================================================================
TestPackageJsonSideEffectsArrayKeepModuleImplicitModule
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index-module.js
console.log("this should be kept");

// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_module_use_module/artifacts.snap [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+## src_entry_js.mjs
+
+```js
+// node_modules/demo-pkg/index-module.js
+console.log('this should be kept');
+
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_keep_module_use_module/artifacts.snap:16]:
```js
================================================================================
TestPackageJsonSideEffectsArrayKeepModuleUseModule
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index-module.js
console.log("this should be kept");

// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_remove/artifacts.snap [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_remove
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_array_remove/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsArrayRemove
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_chain_one/artifacts.snap [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+```js
+// node_modules/d/index.js
+const foo = 123;
+
+// node_modules/b/index.js
+throw 'keep this';
+
+// src/entry.js
+console.log(foo);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_chain_one/artifacts.snap]:
```js
================================================================================
TestPackageJsonSideEffectsFalseIntermediateFilesChainOne
---------- /out.js ----------
// Users/user/project/node_modules/d/index.js
var foo = 123;

// Users/user/project/node_modules/b/index.js
throw "keep this";

// Users/user/project/src/entry.js
console.log(foo);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_diamond/artifacts.snap [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+// node_modules/b1/index.js
+throw 'keep this 1';
+
+// node_modules/b2/index.js
+throw 'keep this 2';
+
+// src/entry.js
+console.log(foo);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_diamond/artifacts.snap]:
```js
================================================================================
TestPackageJsonSideEffectsFalseIntermediateFilesDiamond
---------- /out.js ----------
// Users/user/project/node_modules/d/index.js
var foo = 123;

// Users/user/project/node_modules/b1/index.js
throw "keep this 1";

// Users/user/project/node_modules/b2/index.js
throw "keep this 2";

// Users/user/project/src/entry.js
console.log(foo);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_unused/artifacts.snap [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+expression: content
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_unused
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_intermediate_files_unused/artifacts.snap:12]:
```js
================================================================================
TestPackageJsonSideEffectsFalseIntermediateFilesUnused
---------- /out.js ----------

================================================================================
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_named_import_common_js/artifacts.snap [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+// node_modules/demo-pkg/index.js
+var require_demo_pkg_index = __commonJSMin((exports, module) => {
+	exports.foo = 123;
+	console.log('hello');
+});
+
+// src/entry.js
+var import_demo_pkg_index = __toESM(require_demo_pkg_index());
+console.log(import_demo_pkg_index.foo);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_named_import_common_js/artifacts.snap:22]:
```js
TestPackageJsonSideEffectsFalseKeepNamedImportCommonJS
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
var require_demo_pkg = __commonJS({
  "Users/user/project/node_modules/demo-pkg/index.js"(exports) {
    exports.foo = 123;
    console.log("hello");
  }
});

// Users/user/project/src/entry.js
var import_demo_pkg = __toESM(require_demo_pkg());
console.log(import_demo_pkg.foo);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_named_import_es6/artifacts.snap [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+## src_entry_js.mjs
+
+```js
+// node_modules/demo-pkg/index.js
+const foo = 123;
+console.log('hello');
+
+// src/entry.js
+console.log(foo);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_named_import_es6/artifacts.snap:17]:
```js
================================================================================
TestPackageJsonSideEffectsFalseKeepNamedImportES6
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
var foo = 123;
console.log("hello");

// Users/user/project/src/entry.js
console.log(foo);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_star_import_common_js/artifacts.snap [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+// node_modules/demo-pkg/index.js
+var require_demo_pkg_index = __commonJSMin((exports, module) => {
+	exports.foo = 123;
+	console.log('hello');
+});
+
+// src/entry.js
+var import_demo_pkg_index = __toESM(require_demo_pkg_index());
+console.log(import_demo_pkg_index);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_star_import_common_js/artifacts.snap:22]:
```js
TestPackageJsonSideEffectsFalseKeepStarImportCommonJS
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
var require_demo_pkg = __commonJS({
  "Users/user/project/node_modules/demo-pkg/index.js"(exports) {
    exports.foo = 123;
    console.log("hello");
  }
});

// Users/user/project/src/entry.js
var ns = __toESM(require_demo_pkg());
console.log(ns);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_star_import_es6/artifacts.snap [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+var demo_pkg_index_ns = {};
+__export(demo_pkg_index_ns, {
+	foo:() => foo
+});
+const foo = 123;
+console.log('hello');
+
+// src/entry.js
+console.log(demo_pkg_index_ns);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_keep_star_import_es6/artifacts.snap:23]:
```js
================================================================================
TestPackageJsonSideEffectsFalseKeepStarImportES6
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
var demo_pkg_exports = {};
__export(demo_pkg_exports, {
  foo: () => foo
});
var foo = 123;
console.log("hello");

// Users/user/project/src/entry.js
console.log(demo_pkg_exports);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_no_warning_in_node_modules_issue999/artifacts.snap [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+## src_entry_js.mjs
+
+```js
+// node_modules/demo-pkg/index.js
+console.log('unused import');
+
+// src/entry.js
+console.log('used import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_no_warning_in_node_modules_issue999/artifacts.snap:16]:
```js
================================================================================
TestPackageJsonSideEffectsFalseNoWarningInNodeModulesIssue999
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
console.log("unused import");

// Users/user/project/src/entry.js
console.log("used import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_bare_import_es6/artifacts.snap [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_bare_import_es6
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_bare_import_es6/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsFalseRemoveBareImportES6
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_named_import_es6/artifacts.snap [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_named_import_es6
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_named_import_es6/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsFalseRemoveNamedImportES6
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_star_import_es6/artifacts.snap [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_star_import_es6
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_false_remove_star_import_es6/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsFalseRemoveStarImportES6
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_keep_export_default_expr/artifacts.snap [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+
+## src_entry_js.mjs
+
+```js
+// node_modules/demo-pkg/index.js
+var demo_pkg_index_default = exprWithSideEffects();
+
+// src/entry.js
+console.log(demo_pkg_index_default);
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_keep_export_default_expr/artifacts.snap:16]:
```js
================================================================================
TestPackageJsonSideEffectsKeepExportDefaultExpr
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
var demo_pkg_default = exprWithSideEffects();

// Users/user/project/src/entry.js
console.log(demo_pkg_default);
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_nested_directory_remove/artifacts.snap [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+input_file: crates/rolldown/tests/esbuild/dce/package_json_side_effects_nested_directory_remove
+---
+# Assets
+
+## src_entry_js.mjs
+
+```js
+// src/entry.js
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_nested_directory_remove/artifacts.snap:13]:
```js
================================================================================
TestPackageJsonSideEffectsNestedDirectoryRemove
---------- /out.js ----------
// Users/user/project/src/entry.js
console.log("unused import");
```

File: crates/rolldown/tests/esbuild/dce/package_json_side_effects_true_keep_common_js/artifacts.snap [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+// node_modules/demo-pkg/index.js
+var require_demo_pkg_index = __commonJSMin((exports, module) => {
+	exports.foo = 123;
+	console.log('hello');
+});
+
+// src/entry.js
+var import_demo_pkg_index = __toESM(require_demo_pkg_index());
+console.log('unused import');
+```
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/package_json_side_effects_true_keep_common_js/artifacts.snap:22]:
```js
================================================================================
TestPackageJsonSideEffectsTrueKeepCommonJS
---------- /out.js ----------
// Users/user/project/node_modules/demo-pkg/index.js
var require_demo_pkg = __commonJS({
  "Users/user/project/node_modules/demo-pkg/index.js"(exports) {
    exports.foo = 123;
    console.log("hello");
  }
});

// Users/user/project/src/entry.js
var import_demo_pkg = __toESM(require_demo_pkg());
console.log("unused import");
```

**Review by IWANABETHATGUY — COMMENTED** [review]
duplicated

File: crates/rolldown/tests/esbuild/dce/.package_json_side_effects_array_glob/node_modules/demo-pkg/keep/this/file.js [hunk 21]
```diff
@@ -1 +0,0 @@
-console.log('this should be kept')
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/.package_json_side_effects_array_glob/node_modules/demo-pkg/keep/this/file.js:1]: duplicated

File: crates/rolldown/tests/esbuild/dce/.package_json_side_effects_array_keep/node_modules/demo-pkg/package.json [hunk 22]
```diff
@@ -1,3 +0,0 @@
-{
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/.package_json_side_effects_array_keep/node_modules/demo-pkg/package.json:1]: duplicated

**Review by hyf0 — APPROVED** [review]

**Pull request merged**